src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -974,11 +974,16 @@ HRESULT CProfilerManager::Initialize(
         {
             if (m_attachedClrVersion != ClrVersion_2)
             {
-                IfFailRet(pCallback->Initialize((IUnknown*)(m_pWrappedProfilerInfo.p)));
+                IfFailLog(pCallback->Initialize((IUnknown*)(m_pWrappedProfilerInfo.p)));
             }
             else
             {
-                IfFailRet(pCallback->Initialize((IUnknown*)(m_pRealProfilerInfo.p)));
+                IfFailLog(pCallback->Initialize((IUnknown*)(m_pRealProfilerInfo.p)));
+            }
+
+            if (FAILED(hr))
```

Comment:
If RPH is turned on and fails, but no instrumentation methods are on, do we fail gracefully?

Reply:
I'll double check this, the desired behavior is that the process should have CLRIE running without any instrumentation methods.

Reply:
Confirmed that nothing happens in that case